### ISSUE TITLE
Parameterized tests not reported correctly in 3.16

### ISSUE THREAD

**Issue by bulldetektor** [nunit/nunit3-vs-adapter#694]
**Version info:**
- NUnit3TestAdapter v 3.16.0
- NUnit v 3.12.0
- Test project: netcoreapp2.2 with Microsoft.NET.Test.Sdk v 16.4.0
- Microsoft Visual Studio Enterprise 2019 Version 16.4.2
- TeamCity Enterprise 2019.2 (build 71499)

**Problem:**
When running parameterized tests only the first test is discovered by VS Test Explorer, ReSharper test runner and TeamCity. For instance this test;
```
[Test]
public void Is_applicable_when_no_constraints_are_set(
    [ValueSource(nameof(RandomStrings))] string vehicleGroup,
    [ValueSource(nameof(RandomStrings))] string fuelType,
    [ValueSource(nameof(RandomStrings))] string carBody,
    [Random(5)] int grossWeight)
{
   // removed - not relevant
}
```
...will report only 1 test in v 3.16.0, but on version 3.15.1 there is 625 tests.

Could it be because that the naming of the output tests has changed between v 3.16.0 and 3.15.1? When comparing the outputs in the TeamCity build log I see the following differences:

With 3.16.0:
....Is_applicable_when_no_constraints_are_set
          √ Is_applicable_when_no_constraints_are_set(null,null,null,1895952925)s]

With 3.15.1:
...Is_applicable_when_no_constraints_are_set(null,null,null,912990974)
          √ Is_applicable_when_no_constraints_are_set(null,null,null,912990974)s]		  

As you can see the 3.15.1-tests includes the parameters in the test names, where as in 3.16 they are not.

**Comment by OsirisTerje** [comment]
There is a bug in Resharper.  The adapter now seperates between DisplayName and FullyQualifiedName.  Resharper has used the DisplayName, and this bug is now fixed, see https://youtrack.jetbrains.com/issue/RSRP-477651, and will come in version 2020.1 of Resharper. 

This is also a duplicate of #687

**Comment by bulldetektor** [comment]
Right, but does that explain the build log diffs in TeamCity? The *only* difference between those to builds are the NUnit3TestAdapter version. Same TeamCity version and all other packages are equal.

**Comment by OsirisTerje** [comment]
That was what I saw, and I believe what happens.  The 3.16 uses a FQN of the parent test - without parameters, which is "Is_applicable_when_no_constraints_are_set".  The underlying testcases have all the values added (not their types).  The resharper test runner then only grabs the overlying one.  

I've attached an adapter which is "under work", but which reverts some of these changes.  If you have the possibility to check this one, it could provide some more answers.  

[NUnit3TestAdapter.3.16.2.5-no-dbg.zip](https://github.com/nunit/nunit3-vs-adapter/files/4045662/NUnit3TestAdapter.3.16.2.5-no-dbg.zip)
